docs-source/content/userguide/managing-domains/model-in-image/model-files.md

```diff
@@ -0,0 +1,156 @@
++++
+title = "Model files"
+date = 2020-03-11T16:45:16-05:00
+weight = 30
+pre = "<b> </b>"
+description = "Model file requirements, macros, and loading order."
++++
+
+#### Contents
+
+ - [Introduction](#introduction)
+ - [Sample model file](#sample-model-file)
+ - [Important notes about Model in Image model files](#important-notes-about-model-in-image-model-files)
+ - [Model file naming and loading order](#model-file-naming-and-loading-order)
+ - [Model file macros](#model-file-macros)
+   - [Using secrets in model files](#using-secrets-in-model-files)
+   - [Using environment variables in model files](#using-environment-variables-in-model-files)
+   - [Combining secrets and environment variables in model files](#combining-secrets-and-environment-variables-in-model-files)
+
+#### Introduction
+
+This document describes basic Model in Image model file syntax, naming, and macros. For additional information, see the [WebLogic Deploy Tool](https://github.com/oracle/weblogic-deploy-tooling) documentation.
+
+{{% notice tip %}} The WDT 'discover tool' is particularly useful for generating model files from an existing domain home.
+{{% /notice %}}
+
+#### Sample model file
+
+Here's an example of a model `.yaml` file that defines a WebLogic Administration Server and dynamic cluster.
+
+```
+domainInfo:
+  AdminUserName: '@@SECRET:__weblogic-credentials__:username@@'
+  AdminPassword: '@@SECRET:__weblogic-credentials__:password@@'
+  ServerStartMode: 'prod'
+
+topology:
+  Name: '@@ENV:DOMAIN_UID@@'
+  AdminServerName: "admin-server"
+  Cluster:
+    "cluster-1":
+      DynamicServers:
+        ServerTemplate:  "cluster-1-template"
+        ServerNamePrefix: "managed-server"
+        DynamicClusterSize: 5
+        MaxDynamicClusterSize: 5
+        CalculatedListenPorts: false
+  Server:
+    "admin-server":
+      ListenPort: 7001
+  ServerTemplate:
+    "cluster-1-template":
+      Cluster: "cluster-1"
+      ListenPort: 8001
+```
+
+Some notes about the sample model file:
+ - It includes a WebLogic credentials stanza that is required by Model in Image.
+ - It derives its domain name from the pre-defined environment variable `DOMAIN_UID` but note that this is not required.
+ - For a description of model file macro references to secrets and environment variables, see [Model file macros](#model-file-macros).
+
+#### Important notes about Model in Image model files
+
+- Understand when to use model macros.
+
+  - You can use model macros to reference arbitrary secrets from model files. This is recommended for handling mutable values such as database user names, passwords, and URLs. See [Using secrets in model files](#using-secrets-in-model-files).
+
+    - All password fields in a model should use secret macro. Passwords should not be directly included in property or model files because the files may appear in logs or debugging. 
```

Comment:
Does "secret macro" mean the "@@SECRET" above?

Reply:
Tom AI: "use secret macro" --> "use a secret macro".   Put the above link in all CAPS or at least SECRETS -- USING SECRETS IN MACRO FILES.

Reply:
Done.